Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -284,72 +292,43 @@ extension AWSClient {
-
-        return manageCredential().flatMapThrowing { signer in
-            let awsRequest = try self.createAWSRequest(
-                        operation: operationName,
-                        path: path,
-                        httpMethod: httpMethod,
-                        input: input)
-            return self.createHTTPRequest(awsRequest, signer: signer)
-        }.flatMap { request in
-            return self.invoke(request)
-        }.flatMapThrowing { response in
-            return try self.validate(operation: operationName, response: response)
-        }
-    }
-
-    /// generate a signed URL
-    /// - parameters:
-    ///     - url : URL to sign
-    ///     - httpMethod: HTTP method to use ("GET", "PUT", "PUSH" etc)
-    ///     - expires: How long before the signed URL expires
-    /// - returns:
-    ///     A signed URL
-    public func signURL(url: URL, httpMethod: String, expires: Int = 86400) -> URL {
-        return signer.value.signURL(url: url, method: HTTPMethod(from: httpMethod), expires: expires)
+        return self.credentialProvider.getCredential()
+            .flatMapThrowing { credential in
+                let awsRequest = try self.createAWSRequest(
+                    operation: operationName,
+                    path: path,
+                    httpMethod: httpMethod,
+                    input: input)
+                return self.signRequest(awsRequest, with: credential)
+            }
+            .flatMap { request in
+                return self.invoke(request)
+            }
+            .flatMapThrowing { response in
+                return try self.validate(operation: operationName, response: response)
+            }
     }
 }
 
 // request creator
 extension AWSClient {
 
-    func manageCredential() -> Future<AWSSigner> {
-        #if os(Linux)
-        let signer = self.signer.value
-        if let expiringCredential = signer.credentials as? ExpiringCredential, expiringCredential.nearExpiration() {
-            do {
-                return try MetaDataService.getCredential(httpClient: self.httpClient).map { credential in
-                    let signer = AWSSigner(credentials: credential, name: signer.name, region: signer.region)
-                    self.signer.value = signer
-                    return signer
-                }
-            } catch {
-                // should not be crash
-            }
-        }
-        #endif // os(Linux)
-        return eventLoopGroup.next().makeSucceededFuture(self.signer.value)
-    }
-
-    func createHTTPRequest(_ awsRequest: AWSRequest, signer: AWSSigner) -> AWSHTTPRequest {
-        // if credentials are empty don't sign request
-        if signer.credentials.isEmpty() {
-            return awsRequest.toHTTPRequest()
-        }
+    func signRequest(_ awsRequest: AWSRequest, with credential: Credential) -> AWSHTTPRequest {
+        
+        #warning("we need a way if we don't have credentials here. is that even possible?")
```

Comment:
You removed the ability to created unsigned requests. There are a number of AWS services that allow for unsigned requests (eg CognitoIdentityProvider)